src/script/interpreter.cpp

```diff
@@ -1426,8 +1496,7 @@ bool EvalScript(vector<vector<unsigned char> > &stack,
                     // not used.
                     uint32_t nHashType = GetHashType(vchSig);
                     // BU remember the sighashtype so we can use it to choose when to allow this tx
-                    if (sighashtype)
-                        *sighashtype |= nHashType;
+                    sighashtype |= nHashType;
```

Comment:
sighash type can be null, don't we need to keep the if(sighashtype) statement?

Reply:
no. in the new CScriptMachine class sighashtype is an unsigned char (maybe this should be made into uint8_t) and not a pointer. we are always guaranteed to have some value for it.